feat(tensorflow_frontend): adding mod alias

Gesoti opened the pull request:
<!--
This template will help you to have a meaningful PR, please follow it and do not leave it blank.
-->

# PR Description

<!--
If there is no related issue, please add a short description about your PR.
-->

## Related Issue

<!--
Please use this format to link other issues with their numbers: Close #123
https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword
-->

Close #23532

## Checklist

- [x] Did you add a function?
- [x] Did you add the tests?
- [x] Did you follow the steps we provided?

### Socials:

<!--
If you have Twitter, please provide it here otherwise just ignore this.
-->

Gesoti commented on ivy/functional/frontends/tensorflow/math.py:
```diff
@@ -433,6 +433,11 @@ def minimum(x, y, name=None):
 
+@to_ivy_arrays_and_back
+def mod(x, y, name=None):
+    return ivy.remainder(x, y)
```
I've omitted `check_tensorflow_casting` similarly to the `floormod` function. Is this as expected?

aparajith21 replied:
Tests do pass, but it may be better practice to just add it in!

Gesoti replied:
I've it added! Thank you :)

Gesoti commented:
In tf documentation, this function appears in [raw_ops](https://www.tensorflow.org/api_docs/python/tf/raw_ops/Mod) as well. Do we need to include it in the raw_ops folder here?

Gesoti commented:
Pytests were passing locally. Please advise on next steps.

aparajith21 commented:
Thanks for contributing and sorry for the late review. Tests seem to failing because of dtype errors, you may want to look into the `unsupported_dtypes` decorator and adding dtypes there appropriately. For instance, paddle may not support bfloat16.

Gesoti commented:
> Thanks for contributing and sorry for the late review. Tests seem to failing because of dtype errors, you may want to look into the `unsupported_dtypes` decorator and adding dtypes there appropriately. For instance, paddle may not support bfloat16.

Thank you for the pointers! I've added the unsupported dtypes function wrapper and for now only added the type mentioned here for paddle. Waiting for tests to run to see if there are any other problems. I managed to get the error for `bfloat16` for `paddle` locally which I resolved before pushing upstream. I have a couple of questions:
1. Would it worth adding the unsupported dtypes wrapper to the `remainder` function instead of adding it to the individual `mod` and `floormod` functions?
2.  I've omitted `check_tensorflow_casting` similarly to the `floormod` function. Is this as expected?
3. In tf documentation, this function appears in [raw_ops](https://www.tensorflow.org/api_docs/python/tf/raw_ops/Mod) as well. Do we need to include it in the raw_ops folder here?

aparajith21 commented:
> > Thanks for contributing and sorry for the late review. Tests seem to failing because of dtype errors, you may want to look into the `unsupported_dtypes` decorator and adding dtypes there appropriately. For instance, paddle may not support bfloat16.
> 
> @aparajith21 Thank you for the pointers! I've added the unsupported dtypes function wrapper and for now only added the type mentioned here for paddle. Waiting for tests to run to see if there are any other problems. I managed to get the error for `bfloat16` for `paddle` locally which I resolved before pushing upstream. I have a couple of questions:
> 
> 1. Would it worth adding the unsupported dtypes wrapper to the `remainder` function instead of adding it to the individual `mod` and `floormod` functions?
> 2. I've omitted `check_tensorflow_casting` similarly to the `floormod` function. Is this as expected?
> 3. In tf documentation, this function appears in [raw_ops](https://www.tensorflow.org/api_docs/python/tf/raw_ops/Mod) as well. Do we need to include it in the raw_ops folder here?
> 
> P.S. The tests passed with no new fails being introduced by this PR so I think this might be ready for a review when you get a chance! :)

Great, all looks good to merge. For now, we just add the `unsupported_dtypes` decorator to all functions that need them or where tests fail. For now, it should be sufficient to just include in math.py for this PR, we can always add it there later if necessary. Thanks for the contribution! Will merge once you add the `check_tensorflow_casting` check

aparajith21 commented:
Thanks for contributing!

Gesoti commented:
> > > Thanks for contributing and sorry for the late review. Tests seem to failing because of dtype errors, you may want to look into the `unsupported_dtypes` decorator and adding dtypes there appropriately. For instance, paddle may not support bfloat16.
> > 
> > 
> > @aparajith21 Thank you for the pointers! I've added the unsupported dtypes function wrapper and for now only added the type mentioned here for paddle. Waiting for tests to run to see if there are any other problems. I managed to get the error for `bfloat16` for `paddle` locally which I resolved before pushing upstream. I have a couple of questions:
> > 
> > 1. Would it worth adding the unsupported dtypes wrapper to the `remainder` function instead of adding it to the individual `mod` and `floormod` functions?
> > 2. I've omitted `check_tensorflow_casting` similarly to the `floormod` function. Is this as expected?
> > 3. In tf documentation, this function appears in [raw_ops](https://www.tensorflow.org/api_docs/python/tf/raw_ops/Mod) as well. Do we need to include it in the raw_ops folder here?
> > 
> > P.S. The tests passed with no new fails being introduced by this PR so I think this might be ready for a review when you get a chance! :)
> 
> Great, all looks good to merge. For now, we just add the `unsupported_dtypes` decorator to all functions that need them or where tests fail. For now, it should be sufficient to just include in math.py for this PR, we can always add it there later if necessary. Thanks for the contribution! Will merge once you add the `check_tensorflow_casting` check

Thank you for the pointers! I've added the `check_tensorflow_casting` call. There were some new failures introduced though, from calling `merge_with_upstream.sh`.  Please see below
![image](https://github.com/unifyai/ivy/assets/7903884/6f54d5da-60ce-4353-bf4f-f5a115b7f5ab)

aparajith21 commented:
> > > > Thanks for contributing and sorry for the late review. Tests seem to failing because of dtype errors, you may want to look into the `unsupported_dtypes` decorator and adding dtypes there appropriately. For instance, paddle may not support bfloat16.
> > > 
> > > 
> > > @aparajith21 Thank you for the pointers! I've added the unsupported dtypes function wrapper and for now only added the type mentioned here for paddle. Waiting for tests to run to see if there are any other problems. I managed to get the error for `bfloat16` for `paddle` locally which I resolved before pushing upstream. I have a couple of questions:
> > > 
> > > 1. Would it worth adding the unsupported dtypes wrapper to the `remainder` function instead of adding it to the individual `mod` and `floormod` functions?
> > > 2. I've omitted `check_tensorflow_casting` similarly to the `floormod` function. Is this as expected?
> > > 3. In tf documentation, this function appears in [raw_ops](https://www.tensorflow.org/api_docs/python/tf/raw_ops/Mod) as well. Do we need to include it in the raw_ops folder here?
> > > 
> > > P.S. The tests passed with no new fails being introduced by this PR so I think this might be ready for a review when you get a chance! :)
> > 
> > 
> > Great, all looks good to merge. For now, we just add the `unsupported_dtypes` decorator to all functions that need them or where tests fail. For now, it should be sufficient to just include in math.py for this PR, we can always add it there later if necessary. Thanks for the contribution! Will merge once you add the `check_tensorflow_casting` check
> 
> Thank you for the pointers! I've added the `check_tensorflow_casting` call. There were some new failures introduced though, from calling `merge_with_upstream.sh`. Please see below ![image](https://user-images.githubusercontent.com/7903884/270382897-6f54d5da-60ce-4353-bf4f-f5a115b7f5ab.png)

Those seem unrelated, so it's fine to merge, no need to revert!
